2022/04/formatting-sql-vim-pgformat.md

```diff
@@ -0,0 +1,35 @@
+---
+author: Josh Tolley
+title: "Formatting SQL code with pgFormatter within Vim"
+date: 2022-04-21
+tags:
+- tips
+- open-source
+- tools
+- postgres
+---
+
+Sometimes a little, seemingly simple tip can make a world of difference. I've got enough gray hair these days that it would be pretty easy for me to start thinking I'd seen an awful lot, yet quite frequently when I watch a colleage working in a meeting or a [`tmux`](https://github.com/tmux/tmux) session or somewhere, I learn some new and simple thing that makes my life demonstrably easier. [Luca Ferrari's recent post](https://fluca1978.github.io/2022/04/13/EmacsPgFormatter.html) about using pgFormatter in Emacs reminded me that I'd meant for a while to share that essentially the same thing works in Vim. I don't especially want to get involved an editor war, and offer the following only in the spirit of friendly cooperation for the vim users out there.
+
+As Luca mentioned, pgFormatter is a convenient way to make SQL queries readable, automatically. It's easy enough to feed it some SQL, and get a nice-looking result as output:
+
+```
+josh@igtre:~$ pg_format < create_outbreaks.sql
+INSERT INTO outbreak                      
+SELECT                              
+    nextval('outbreak_id'::regclass),
+    extract('year' FROM now())::text || '-' || nextval('outbreak_number_seq')::text, --number
+    (                        
+        SELECT  
+            first_name
+        FROM person TABLESAMPLE BERNOULLI (10)
+LIMIT 1), -- name
```

Comment:
Does it really put this `LIMIT 1),` unindented like that? Seems like a bug if so.

Reply:
It does, and I agree. I chalk that up to one of the design decisions I haven't chosen to bother trying to improve.